bin/crew

```diff
@@ -1587,8 +1583,20 @@ def remove(pkg_name)
         end
       end
 
+      # Iterate over each filepath, extracting every level of parent directories
+      directories = []
+      filepaths.uniq.each do |directory|
+        current_directory = directory
+        loop do
+          directories << current_directory
+          break if current_directory == CREW_PREFIX
+          current_directory = File.dirname(current_directory)
+        end
+      end
+      directories.uniq!.map!(&:to_s)
```

Comment:
Isn't this reinventing the wheel?

Reply:
By doing this at runtime, we are able avoid creating static lists of directories in addition to lists of files and thus save space.

Reply:
What if we added directories to the filelists and removed everything in one fell swoop?  The directory would have to be the last entry after every file in that directory, however.

Reply:
As in, added directories to our existing filelists in `manifests`?

Reply:
Yes, that way we could loop through all filelist entries and do a wholesale delete.

Reply:
I'm open to the idea-- but unless we are going to regenerate the filelist for every package, we'll still need to keep some code around to handle legacy cases.

Reply:
Although, we'd need some way to ensure that we remove directories last.

Reply:
That was the intention of the .directorylist file.  You could append the directory list to the filelist and still have only one filelist per package.

Reply:
True-- that does mean we don't get a bunch of the space savings, though.